### DISCUSSION TITLE
Wsparcie dla czujnika BME680

### DISCUSSION THREAD

**Discussion by wiuwiu** [hackerspace-silesia/Smogomierz#136] [Ideas]
Z ciekawych rzeczy fajnie by było mieć obsługę MQTT w Smogomierzu by móc przesyłać dane bezpośrednio do lokalnego systemu inteligentnego domu. 
Kolejną ciekawą rzeczą jest obsługa czujnika BME680 z pomiarem VOC / gazów. Bosh udostępnia firmware dla ESP8266 do obliczania indeksu jakości powietrza IAQ. Link: [https://botland.com.pl/pl/czujniki-gazow/10872-adafruit-bme680-czujnik-temperatury-wilgotnosci-cisnienia-i-gazu.html](url)

**Comment by bfaliszek** [comment]
Zakładam, że Smogomierze są umieszczane raczej na zewnątrz, więc pomiar IAQ (Indoor air quality), plus to co podaje BME680 to nie jest wartość VOC, tylko rezystancja wszystkich gazów. Myślałem o dodaniu CCS811, który podaje właśnie VOC i CO2, ale on nie podaje wartości absolutnych i każdy z nich wymaga kalibracji, więc raczej za dużo z tym zachodu. Chciałbym dodać coś do pomiaru NO2 ale większość mierników nie radzi sobie z niskimi wartościami, a te które podają wartość od 0, kosztują majątek.

**Comment by wiuwiu** [comment]
Wzięte z opisu czujnika: 

> Czujnik BME680 zawiera mały czujnik MOX. Ogrzewany tlenek metalu zmienia rezystancje w oparciu o lotne związki organiczne (VOC) znajdujące się w powietrzu, dzięki czemu może wykrywać gazy i alkohole, takie jak etanol, alkohol i tlenek węgla. Może być także wykorzystywany do wykonywania pomiarów czystości powietrza. 

> 
W przypadku CCS811 można by oprzeć się nie tyle na samych wartościach co na ich zmianie. Wówczas kalibracja nie byłaby potrzeba a każdy by widział, że jeżeli wartości nagle strzelają w górę to "wiedz, że coś się dzieje" :)

**Comment by bfaliszek** [comment]
W 2.6.0 dodałem wstępnie wsparcie dla MQTT, tylko nie miałem za bardzo jak przetestować, więc będę wdzięczny na info czy działa – https://github.com/hackerspace-silesia/Smogomierz/tree/2.6.0

**Comment by hejsiri** [comment]
Wygląda na to że coś tam się pojawia w brokerze...

![IMG_1806](https://user-images.githubusercontent.com/37380727/54878720-d32f4d00-4e30-11e9-9ac7-3935934f2d72.jpeg)
 Możnaby ewentualnie dodać w składni topiców nazwę urządzenia, np. tak:

Smogomierz-8421EF/sensor/temperature

gdzie "Smogomierz-8421EF" to nazwa urządzenia. Przy dwóch smogomierzach na tym samym brokerze MQTT jest konflikt, a wyniki nadpisują się pod takim samym topikiem.

**Comment by bfaliszek** [comment]
@mientki dzięki za info. Dodam to w kolejnej aktualizacji.

**Comment by bfaliszek** [comment]
Jest już 2.6.2 - https://github.com/hackerspace-silesia/Smogomierz/releases/tag/v2.6.2 poprawiłem m.in. tam wsparcie dla MQTT

**Comment by hejsiri** [comment]
Mam 2 smogomierze w sieci:
1) Nazwa urządzenia: Smogomierz-8421EF 
2) Nazwa urządzenia: Smogomierz-0A5017 

Na brokerze MQTT pojawiają zupełnie inne nazwy:

![IMG_1998](https://user-images.githubusercontent.com/37380727/55682145-b1a18b80-592f-11e9-88a6-45656eb5267a.png)


Z czego to może wynikać?

**Comment by bfaliszek** [comment]
Nazwa urządzenie jest trochę inaczej generowana. To co jest w brokerze to ChipID, który jest podany na stronie Konfiguracji przy opcji włączenia luftdaten. 
ChipID jest bardziej unikatowe niż nazwa urządzenia. Zwłaszcza, że nazwę urządzenia możesz zmienić. ChipID nie możesz zmienić w ustawieniach. To co jest wyświetlane w brokerze to właśnie ChipID
